src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/authentication/RequireGroupAuthenticator.java

```diff
@@ -0,0 +1,163 @@
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.sessions.AuthenticationSessionModel;
+
+
+/**
+ * Simple {@link Authenticator} that checks of a user is member of a given {@link GroupModel Group}.
+ */
+public class RequireGroupAuthenticator implements Authenticator {
+
+    /**
+     * Logger variable.
+     */
+    private static final Logger LOGGER = Logger.getLogger(RequireGroupAuthenticator.class.getName());
+
+    /**
+     * This implementation is not intended to be overridden.
+     */
+    @Override
+    public void authenticate(final AuthenticationFlowContext context) {
+
+        RealmModel realm = context.getRealm();
+        UserModel user = context.getUser();
+        AuthenticationSessionModel authenticationSession = context.getAuthenticationSession();
+        ClientModel client = authenticationSession.getClient();
+        String clientId = client.getClientId();
+        String groupName = client.getAttribute("groups");
+        String logPrefix = "UDS_GROUP_PROTECTION_AUTHENTICATE_" + authenticationSession.getParentSession().getId();
+
+        if (user != null) {
+            LOGGER.infof("%s user %s / %s", logPrefix, user.getId(), user.getUsername());
+        } else {
+            LOGGER.warnf("%s invalid user", logPrefix);
+
+        }
+        LOGGER.infof("%s client %s / %s", logPrefix, clientId, client.getName());
+        LOGGER.infof("%s %s client has group attribute %s", logPrefix, client.getName(), groupName);
+
+        // Check for a valid match
+        if (groupName.length() > 0) {
+            LOGGER.infof("%s %s client has group attribute %s", logPrefix, client.getName(), groupName);
+            Optional<GroupModel> foundGroup = getGroupByName(groupName, realm);
+
+            if (foundGroup.isPresent()) {
+                checkIfUserIsAuthorized(context, realm, user, logPrefix, foundGroup.get());
+            } else {
+                LOGGER.warnf("%s Groups attribute (%s) failed to find matching group for %s client - the group does not exist in realm.", logPrefix, groupName, client.getName());
+                // This failure (group not existing) is surfaced the same to the user as if the user is not a member of the group
+                // Perhaps should be separate error
+                LOGGER.warnf("test adding a line");
+                context.failure(AuthenticationFlowError.INVALID_CLIENT_SESSION);
+            }
+        } else {
+            LOGGER.infof("%s No groups detected for %s client", logPrefix, client.getName());
+            success(context, user);
+        }
+    }
+
+    private Optional<GroupModel> getGroupByName(final String groupName, final RealmModel realm) {
+        return realm.getGroupsStream()
+            .filter(group -> group.getName().equals(groupName))
+            .findFirst();
```

Comment:
How does this work with nested groups? Is this matching on the short group name or the full path?